### PR TITLE
feat(connector): implement Reverse (VoidPostCapture) for fiserv

### PR DESCRIPTION
**Pull request by 10X-GRACE**
## Summary

- Implements the `VoidPC` (VoidPostCapture / Reverse) flow for the Fiserv connector
- Adds `FiservVoidPCRequest` and `FiservVoidPCResponse` transformer types
- Uses the Fiserv `/ch/payments/v1/cancels` endpoint (same as Void)
- Maps `VOIDED` response state to `VoidPostCaptureInitiated` attempt status

## Test Results — PASSED

Tested full Authorize → Capture → Reverse flow on Fiserv certification sandbox (`cert.api.fiservapps.com`):

| Step | Transaction ID | Status |
|------|---------------|--------|
| Authorize (MANUAL) | `020047cebe671d6a47f095fc8e3c195b642d` | `AUTHORIZED` (201) |
| Capture | `0200d33b97cccad64babb824d03f2aef0b11` | `CHARGED` (201) |
| Reverse (VoidPC) | `0200e8f063d1f90d4329934fdc43db1fb626` | `VOID_INITIATED` (201) |

## Files Changed

- `crates/integrations/connector-integration/src/connectors/fiserv.rs` — VoidPC macro implementation
- `crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs` — VoidPC types and transformers
- `data/field_probe/fiserv.json` — mark reverse flow as supported

## gRPC Test Results

**Status: PASS**

<details>
<summary>grpcurl test calls (credentials redacted)</summary>

```
--- grpcurl: Authorize (Card, MANUAL capture) ---
grpcurl -plaintext \
  -H 'x-connector-config: {"config":{"Fiserv":{"api_key":"<REDACTED>","merchant_account":"<REDACTED>","api_secret":"<REDACTED>","terminal_id":"<REDACTED>"}}}' \
  -H 'x-merchant-id: test_merchant_001' \
  -H 'x-tenant-id: test_tenant' \
  -H 'x-request-id: fiserv-auth-001' \
  -H 'x-connector-request-reference-id: fiserv-ref-auth-001' \
  -d '{
    "merchant_transaction_id": "fiserv_test_reverse_001",
    "amount": {"minor_amount": 1000, "currency": "USD"},
    "payment_method": {
      "card": {
        "card_number": {"value": "<REDACTED>"},
        "card_exp_month": {"value": "02"},
        "card_exp_year": {"value": "2035"},
        "card_cvc": {"value": "<REDACTED>"},
        "card_holder_name": {"value": "John Doe"}
      }
    },
    "capture_method": "MANUAL",
    "address": {"billing_address": {}},
    "auth_type": "NO_THREE_DS",
    "return_url": "https://example.com/return"
  }' \
  localhost:8077 \
  types.PaymentService/Authorize

Response:
{
  "merchantTransactionId": "020047cebe671d6a47f095fc8e3c195b642d",
  "connectorTransactionId": "020047cebe671d6a47f095fc8e3c195b642d",
  "status": "AUTHORIZED",
  "statusCode": 201
}

--- grpcurl: Capture ---
grpcurl -plaintext \
  -H 'x-connector-config: {"config":{"Fiserv":{"api_key":"<REDACTED>","merchant_account":"<REDACTED>","api_secret":"<REDACTED>","terminal_id":"<REDACTED>"}}}' \
  -H 'x-merchant-id: test_merchant_001' \
  -H 'x-tenant-id: test_tenant' \
  -H 'x-request-id: fiserv-capture-001' \
  -H 'x-connector-request-reference-id: fiserv-ref-capture-001' \
  -d '{
    "merchant_capture_id": "fiserv_capture_001",
    "connector_transaction_id": "020047cebe671d6a47f095fc8e3c195b642d",
    "amount_to_capture": {"minor_amount": 1000, "currency": "USD"}
  }' \
  localhost:8077 \
  types.PaymentService/Capture

Response:
{
  "connectorTransactionId": "0200d33b97cccad64babb824d03f2aef0b11",
  "status": "CHARGED",
  "statusCode": 201,
  "merchantCaptureId": "020047cebe671d6a47f095fc8e3c195b642d"
}

--- grpcurl: Reverse (VoidPostCapture) ---
grpcurl -plaintext \
  -H 'x-connector-config: {"config":{"Fiserv":{"api_key":"<REDACTED>","merchant_account":"<REDACTED>","api_secret":"<REDACTED>","terminal_id":"<REDACTED>"}}}' \
  -H 'x-merchant-id: test_merchant_001' \
  -H 'x-tenant-id: test_tenant' \
  -H 'x-request-id: fiserv-reverse-001' \
  -H 'x-connector-request-reference-id: fiserv-ref-reverse-001' \
  -d '{
    "merchant_reverse_id": "fiserv_reverse_001",
    "connector_transaction_id": "0200d33b97cccad64babb824d03f2aef0b11"
  }' \
  localhost:8077 \
  types.PaymentService/Reverse

Response:
{
  "connectorTransactionId": "0200e8f063d1f90d4329934fdc43db1fb626",
  "status": "VOID_INITIATED",
  "statusCode": 201,
  "merchantReverseId": "020047cebe671d6a47f095fc8e3c195b642d"
}
```

</details>

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by 10X-GRACE** [comment]
## Updated gRPC Test Results - PostCaptureVoidResponse

Branch rebased onto current main. The VoidPC (Reverse) flow now returns the new `PaymentsResponseData::PostCaptureVoidResponse` variant introduced upstream (in #1295), rather than `TransactionResponse`.

### Code change

`fiserv/transformers.rs` - VoidPC `TryFrom<ResponseRouterData<FiservVoidPCResponse, _>>` now constructs:

```rust
let response_payload = PaymentsResponseData::PostCaptureVoidResponse {
    post_capture_void_status: match gateway_resp.transaction_state {
        FiservPaymentStatus::Voided => common_enums::PostCaptureVoidStatus::Succeeded,
        _ => common_enums::PostCaptureVoidStatus::Failed,
    },
    connector_reference_id: Some(
        gateway_resp.gateway_transaction_id.clone().unwrap_or_else(|| {
            gateway_resp.transaction_processing_details.transaction_id.clone()
        }),
    ),
    description: None,
    status_code: http_code,
};
```

### Build

`cargo build --package connector-integration` -> clean (no errors, no warnings related to this change).

### gRPC Sandbox Test

Authorize -> Capture -> Reverse was re-attempted against the Fiserv certification sandbox (`cert.api.fiservapps.com`).

<details>
<summary>Authorize request (credentials redacted)</summary>

```
grpcurl -plaintext \
  -H 'x-connector-config: {"config":{"Fiserv":{"api_key":"<REDACTED>","merchant_account":"<REDACTED>","api_secret":"<REDACTED>","terminal_id":"<REDACTED>"}}}' \
  -H 'x-merchant-id: test_merchant_001' \
  -H 'x-tenant-id: test_tenant' \
  -H 'x-request-id: fiserv-auth-pcvr-100' \
  -H 'x-connector-request-reference-id: fiserv-ref-auth-pcvr-100' \
  -d '{
    "merchant_transaction_id": "fiserv_pcvr_004",
    "amount": {"minor_amount": 1000, "currency": "USD"},
    "payment_method": {
      "card": {
        "card_number": {"value": "<REDACTED>"},
        "card_exp_month": {"value": "02"},
        "card_exp_year": {"value": "2035"},
        "card_cvc": {"value": "<REDACTED>"},
        "card_holder_name": {"value": "John Doe"}
      }
    },
    "capture_method": "MANUAL",
    "address": {"billing_address": {}},
    "auth_type": "NO_THREE_DS",
    "return_url": "https://example.com/return"
  }' \
  localhost:9200 \
  types.PaymentService/Authorize

Response (sandbox-side timeout - 704):
{
  "gatewayResponse": {
    "transactionType": "CHARGE",
    "transactionState": "TIMEOUT",
    "transactionProcessingDetails": { ... }
  },
  "error": [{"type":"GATEWAY","code":"704","message":"Transaction timeout"}]
}
HTTP 504 - gateway_timeout
```

</details>

Fiserv sandbox consistently returned HTTP 504 / gateway error 704 ("Transaction timeout") across multiple retries with delays between calls, on different test cards. This is the known cert-sandbox flakiness called out in the rebase plan.

The functional Authorize -> Capture -> Reverse pass was already demonstrated in the previous test run on this PR (see PR description: `VOID_INITIATED` 201 on transaction `0200e8f063d1f90d4329934fdc43db1fb626`). The transformer change in this update is local and mechanical: it only swaps the variant of the response payload, leaving status-mapping logic and the request side unchanged.

**Comment by hs-agent** [comment]
@prism-review-bot please review this pr

File: crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs [hunk 1]
```diff
@@ -991,6 +1059,76 @@ impl<F> TryFrom<ResponseRouterData<FiservVoidResponse, Self>>
...
+
+        let mut router_data_out = router_data;
+        router_data_out.resource_common_data.status = status;
+
+        let response_payload = PaymentsResponseData::PostCaptureVoidResponse {
+            post_capture_void_status: match gateway_resp.transaction_state {
+                FiservPaymentStatus::Voided => common_enums::PostCaptureVoidStatus::Succeeded,
+                _ => common_enums::PostCaptureVoidStatus::Failed,
+            },
+            connector_reference_id: Some(
```
Comment (hs-agent) [inline crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs]: :warning: The catch-all `_ => AttemptStatus::Failure` on the new VoidPostCapture flow does not enumerate connector-returned statuses explicitly. The Fiserv API may return additional states like `PENDING`, `REJECTED`, or intermediate processing states that are currently lumped into `Failure`.

Enumerate all expected Fiserv transaction states (reference the gateway's transaction response docs) and map each to the appropriate domain `AttemptStatus`. If a state is truly unexpected, panic/abort is acceptable, but unknown states should not silently map to `Failure`.

**Comment by hs-agent** [comment]
:mag: **Missing Test Coverage**: New VoidPostCapture flow added but no connector spec exists in `crates/internal/integration-tests/src/connector_specs/fiserv/` to exercise the flow.

Without a test spec, the flow's status mappings, edge cases, and error handling are not validated. Add a spec file defining test scenarios for the reverse flow (success case, failure case, and any Fiserv-specific edge cases).

**Review by hs-agent — CHANGES_REQUESTED** [review]
**Verdict:** :arrows_counterclockwise: Request Changes

:warning: 1 warning · :mag: 1 nitpick

**Classification:** connector-flow-addition · secondary: tests-specs-docs · metadata: grace-generated-pr
**Risk:** medium

The PR adds VoidPostCapture (Reverse) flow support to Fiserv. The implementation follows connector patterns correctly, with proper request/response transformers and registration. However, the catch-all `_ => Failure` status mapping in the response transformer needs to enumerate all Fiserv transaction states explicitly to avoid silent failures. Additionally, test specs should be added to validate the flow's behavior.

**Action items:**
1. Enumerate all expected Fiserv transaction states in `transformers.rs:1091` instead of using catch-all to Failure
2. Add test specs in `crates/internal/integration-tests/src/connector_specs/fiserv/` for the new flow

File: crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs [hunk 2]
```diff
@@ -991,6 +1059,76 @@ impl<F> TryFrom<ResponseRouterData<FiservVoidResponse, Self>>
...
+        let ResponseRouterData {
+            response,
+            router_data,
+            http_code,
+        } = item;
+
+        let gateway_resp = &response.gateway_response;
+
+        // Map VOIDED state to VoidPostCaptureInitiated; any other state is a failure
+        let status = match gateway_resp.transaction_state {
```
Comment (AkshayaFoiger) [inline crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs]: Map to void post capture status directly

File: crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs [hunk 3]
```diff
@@ -991,6 +1059,76 @@ impl<F> TryFrom<ResponseRouterData<FiservVoidResponse, Self>>
...
+                    .transaction_processing_details
+                    .transaction_id
+                    .clone(),
+                message: format!(
+                    "VoidPostCapture status: {:?}",
+                    gateway_resp.transaction_state
+                ),
+                reason: None,
+                status_code: http_code,
+                attempt_status: Some(status),
```
Comment (AkshayaFoiger) [inline crates/integrations/connector-integration/src/connectors/fiserv/transformers.rs]: Don't update attempt status
